articles/vercel-preview/index.mdx

```diff
@@ -1,37 +1,114 @@
+- **Reducing CI cost:** Eliminating the need to set up a second environment to run the tests.
+- **Increasing test reliability**: Visual testing on a production-like environment.
+- **Increasing test reliability**: Visual testing on a production-like environment.
 
-## How to run Argos on Vercel Preview
+**Note:** This method is optimal for repeatable tests that don't significantly alter the environment, as certain tests (e.g. those involving user deletion) may not be applicable.
 
-### 1. Set up Argos
+## Getting Started with Argos: Setup Guide
 
-Start by following the Q[Quickstart guide](https://argos-ci.com/docs) to integrate Argos with Playwright. This foundational step ensures that Argos is ready to run visual tests during your CI process.
+### Prerequisites
 
-### 2. Run tests on your Vercel Preview
+Before integrating Argos into Vercel Preview for visual testing, I have:
 
-Adapt your CI workflow with this GitHub Actions configuration to enable Argos visual testing on Vercel Previews:
+1. Set up a [Next.js](https://nextjs.org/) project with [Playwright](https://playwright.dev).
+2. Integrated Playwright tests into CI/CD.
+3. Deployed the project on Vercel.
+
+### 1. Project Importation to Argos
+
+[Signed up into Argos](https://app.argos-ci.com/signup) and import my project. In response, Argos generated an `ARGOS_TOKEN` for my repository that I saved it for later usage.
+
+![Argos GitHub token](https://dev-to-uploads.s3.amazonaws.com/uploads/articles/wra1891953mfv4kt6tcs.png)
+
+### 2. Package Installation
+
+```sh
+npm install --save-dev @argos-ci/playwright
+```
+
+### 3. Setup Argos in Playwright config file
+
+Update the `playwright.config.ts` file:
+
+```ts
+import { defineConfig } from "@playwright/test";
+
+export default defineConfig({
+  reporter: [
+    process.env.CI ? ["dot"] : ["list"],
+    [
+      "@argos-ci/playwright/reporter",
+      {
+        uploadToArgos: !!process.env.CI,
+        token: process.env.ARGOS_TOKEN,
+      },
+    ],
+  ],
+  use: {
+    trace: "on-first-retry",
+    screenshot: "only-on-failure",
+  },
+});
+```
+
+### 4. Screenshot my app
+
+I utilized the [Screenshot pages script](https://argos-ci.com/docs/screenshot-pages-script) from Argos's documentation for capturing my application's homepage. This script is flexible for adding more pages in the future.
+
+```ts
+import { argosScreenshot } from "@argos-ci/playwright";
+import { test } from "@playwright/test";
+
+const baseUrl = "http://localhost:3000";
+const pages = [
+  { name: "homepage", path: "/" },
+  // { name: "faq", path: "/faq" },
+  // { name: "contact", path: "/contact-us" },
+  // { name: "pricing", path: "/pricing" },
+];
+
+test("screenshot pages", async ({ page }, workerInfo) => {
+  for (const { name, path } of pages) {
+    const browserName = workerInfo.project.name;
+    await page.goto(`${baseUrl}${path}`);
+    await argosScreenshot(page, `${name}-${browserName}`);
+  }
+});
```

Comment:
It would be more elegant to iterate and generate one test per page, we should try that and update docs if possible.